File: Ruby/problem071.rb
class Problem071
  class << self
    def rand5
      # Get 50 random numbers
      candidates = (1..50).map { |n| Problem071.rand7 }
      # Pick the one I want from the filtered array.
      (candidates.filter { |n| n <= 5 }).sample
    end

    def rand7
      (1..7).to_a.sample
    end
  end
end

File: Lua/problem071.lua
-- You need to initialize a table for exporting the module.
problem071 = {}

local function rand7()
    return math.random(7)
end

-- The #size method only counts INTEGER values in the array.
-- Since we're winging a set with the table data type, we need
-- another function to properly count the elements.
local function tableLength(T)
    local count = 0
    for _ in pairs(T) do
        count = count + 1
    end
    return count
end

function problem071.rand5()
    local out = rand7()
    while out > 5 do
        out = rand7()
    end
    return out
end

-- Random needs to be seeded outside of the generator function,
-- otherwise it will be repeating the same result.
math.randomseed(os.time())

return problem071
